### PR TITLE
Fixed `_field_types` for python > 3.9

### PR DESCRIPTION
**Pull request by aresler**
According to the [documentation](https://docs.python.org/3/library/typing.html#typing.NamedTuple):

> Changed in version 3.9: Removed the _field_types attribute in favor of the more standard __annotations__ attribute which has the same information.

### CODE REVIEW

**Comment by AlexWaygood** [comment]
The check in `tests/check_new_syntax.py` needs to be changed — it's not your fault!

**Comment by aresler** [comment]
Sorry for all the pushing. Indeed, the test may need some changes. I left what I think is the shortest version that covers all cases with Python < 3.8, between 3.8 and 3.9, and > 3.9. 

Or, take the previous commit with all checks passed, but it doesn't cover all cases accurately.

>Changed in version 3.8: The _field_types and __annotations__ attributes are now regular dictionaries instead of instances of OrderedDict.

>Changed in version 3.9: Removed the _field_types attribute in favor of the more standard __annotations__ attribute which has the same information.

File: stdlib/typing.pyi [hunk 1]
```diff
@@ -682,7 +682,10 @@ def cast(typ: object, val: Any) -> Any: ...
 # Type constructors
 
 class NamedTuple(Tuple[Any, ...]):
-    _field_types: collections.OrderedDict[str, Type[Any]]
+    if sys.version_info < (3, 8):
+        _field_types: collections.OrderedDict[str, Type[Any]]
+    elif sys.version_info < (3, 9):
+        _field_types: dict[str, Type[Any]]
```
Comment (Akuli) [inline stdlib/typing.pyi:688]: Nit:

```suggestion
        _field_types: collections.OrderedDict[str, type]
    elif sys.version_info < (3, 9):
        _field_types: dict[str, type]
```

**Review by Akuli — APPROVED** [review]

**Comment by Akuli** [comment]
Thanks!

**Pull request merged**